test/test_pipeline/components/setup/test_setup_traditional_models.py

```diff
@@ -0,0 +1,123 @@
+    def test_get_set_config_space(self, dataset_properties):
+        """Make sure that we can setup a valid choice in the encoder
+        choice"""
+        model_choice = ModelChoice(dataset_properties)
+        cs = model_choice.get_hyperparameter_search_space(dataset_properties=dataset_properties)
+
+        # Make sure that all hyperparameters are part of the search space
+        assert sorted(cs.get_hyperparameter('__choice__').choices) == sorted(list(model_choice.get_components().keys()))
+
+        # Make sure we can properly set some random configs
+        # Whereas just one iteration will make sure the algorithm works,
+        # doing five iterations increase the confidence. We will be able to
+        # catch component specific crashes
+        for i in range(5):
+            config = cs.sample_configuration()
+            config_dict = copy.deepcopy(config.get_dictionary())
+            model_choice.set_hyperparameters(config)
+
+            assert model_choice.choice.__class__ == model_choice.get_components()[config_dict['__choice__']]
+
+            # Then check the choice configuration
+            selected_choice = config_dict.pop('__choice__', None)
+            for key, value in config_dict.items():
+                # Remove the selected_choice string from the parameter
+                # so we can query in the object for it
+                key = key.replace(selected_choice + ':', '')
+                assert key in vars(model_choice.choice)['config']
+                assert value == model_choice.choice.__dict__['config'][key]
+
+
+@pytest.mark.parametrize("fit_dictionary_tabular", ['classification_categorical_only',
+                                                    'classification_numerical_only',
+                                                    'classification_numerical_and_categorical',
+                                                    "regression_numerical_only",
+                                                    "regression_categorical_only",
+                                                    "regression_numerical_and_categorical"
+                                                    ], indirect=True)
+class TestTraditionalModels:
+    def test_model_fit_predict_score(self, traditional_learner, fit_dictionary_tabular):
+
+        if len(fit_dictionary_tabular['dataset_properties']['numerical_columns']) == 0 and traditional_learner == 'knn':
+            pytest.skip("knn can not work with categorical only data")
+
+        model = TabularTraditionalModel(traditional_learner=traditional_learner)
+
+        blockPrint()
+        model.fit(X=fit_dictionary_tabular)
+        enablePrint()
+
+        assert isinstance(model.fit_output, dict)
+        assert 'val_preds' in model.fit_output.keys()
+        assert isinstance(model.fit_output['val_preds'], list)
+        assert len(model.fit_output['val_preds']) == len(fit_dictionary_tabular['val_indices'])
+        if model.model.is_classification:
```

Comment:
Can you please add a unit test that makes sure that is_classifiaction what set properly? Like I was not able to find where in the code we make sure that is properly setup up...

Reply:
https://github.com/ravinkohli/Auto-PyTorch/blob/1f0e35d399ff41b0ceb12ff8ba952714fb6d498d/autoPyTorch/pipeline/components/setup/traditional_ml/traditional_learner/base_traditional_learner.py#L68